internal/dataplane/deckgen/consumer_cmp.go

```diff
@@ -0,0 +1,50 @@
+package deckgen
+
+import (
+	"strings"
+
+	"github.com/kong/go-database-reconciler/pkg/file"
+)
+
+type fConsumerByUsernameAndCustomID []file.FConsumer
+
+func (f fConsumerByUsernameAndCustomID) Len() int      { return len(f) }
+func (f fConsumerByUsernameAndCustomID) Swap(i, j int) { f[i], f[j] = f[j], f[i] }
+func (f fConsumerByUsernameAndCustomID) Less(i, j int) bool {
+	if f[i].Username == nil && f[j].Username != nil {
+		return true
+	}
+	if f[i].Username != nil && f[j].Username == nil {
+		return false
+	}
+	if f[i].Username != nil && f[j].Username != nil {
+		switch cmp := strings.Compare(*f[i].Username, *f[j].Username); cmp {
+		case -1:
+			return true
+		case 1:
+			return false
+		case 0:
+			break
+		}
+	}
+
+	// Both usernames are equal, compare custom_id.
```

Comment:
```suggestion
	// Both usernames are empty or equal, compare custom_id.
```
Should we mention that we would compare `custom_id` if both `username` are empty in the comment here?

Reply:
Technically if they are both empty then they are equal :) But I'm OK with adding this note in the comment.